src/common/dto/view/CreateView.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsEnum,
+  IsString,
+  MinLength,
+} from 'class-validator';
+import { COURSE_TABLE_COLUMN } from 'common/constants';
+
+/**
+ * Create View DTO
+ *
+ * Defines the data shape for creating a new [[View]] entry in the database.
+ */
+export class CreateViewDTO {
+  @ApiProperty({
+    description: 'Array of column names for this view',
+    example: COURSE_TABLE_COLUMN.AREA,
+    enum: COURSE_TABLE_COLUMN,
+  })
+  @IsEnum(COURSE_TABLE_COLUMN, { each: true })
+  public columns: COURSE_TABLE_COLUMN[] = [];
+
+  @ApiProperty({
+    example: 'No semester data',
+    description: 'A descriptive name for this custom view',
+  })
+  @IsString()
+  @MinLength(1)
+  public name: string;
```

Comment:
We have a decorator that trims the string (`@Transform(trimString)`) that's being used in the `CreateCourse` dto that could be used here along with for the other `name` properties in order to sanitize the text input from the user.

Reply:
Fixed that in b3f67944610c487c6f84d6e08782e15138818a54